runtime/interpreter/interpreter.go

```diff
@@ -3671,6 +3671,57 @@ func (interpreter *Interpreter) storageAccountPaths(addressValue AddressValue, g
 	return interpreter.accountPaths(addressValue, getLocationRange, common.PathDomainStorage, PrimitiveStaticTypeStoragePath)
 }
 
+func (interpreter *Interpreter) iterOverStorageDomain(addressValue AddressValue, domain common.PathDomain, pathType sema.Type) *HostFunctionValue {
```

Comment:
Maybe improve naming here: starting the name of the function with a verb makes it sound like it performs the iteration, but it does not, it only returns a function. A common naming scheme is e.g.

```suggestion
func (interpreter *Interpreter) newStorageIterationFunction(addressValue AddressValue, domain common.PathDomain, pathType sema.Type) *HostFunctionValue {
```